include/dali/core/mm/cuda_vm_resource.h

```diff
@@ -265,8 +265,8 @@ class cuda_vm_resource : public memory_resource<memory_kind::device> {
     size_t                  block_size = 0;
   };
 
-  SmallVector<va_region, 8> va_regions_;
-  SmallVector<cuvm::CUMemAddressRange, 8> va_ranges_;
+  std::vector<va_region> va_regions_;
+  std::vector<cuvm::CUMemAddressRange> va_ranges_;
```

Comment:
Is there any perf difference?

Reply:
No I see https://github.com/NVIDIA/DALI/pull/3117#discussion_r665267246. Ok

Reply:
There shouldn't be any. Increasing the VA size is very rare and it's a big event in itself, taking much more time than simple heap reallocation.